### ISSUE TITLE
Clixon compile error "undeclared identifier SOCK_NONBLOCK" on macOS

### ISSUE THREAD

**Issue by brunorijsman** [clicon/clixon#540]
Clixon has a compile error on macOS.

Clixon version:
Branch: master
Commit hash: 9da4939ee09f7c7b43087390297e68f4bc76304e
Clone date: 13-Aug-2024

MacOS version: Ventura 13.3.1

Reproduce steps:
```
git clone https://github.com/clicon/clixon.git
cd clixon
./configure
make
```

Error message:
```
gcc -I. -I../../lib/src -I../../lib -I../../include -I../..  -D__PROGRAM__=\"clixon_restconf\" -DHAVE_CONFIG_H  -fPIC -O2 -Wall  -c restconf_main_native.c
restconf_main_native.c:764:34: error: use of undeclared identifier 'SOCK_NONBLOCK'
                                 SOCK_NONBLOCK, /* Also 0 is possible */
```

Temporary workaround for now: in line 171, of restconf_main_native.c comment out the line:
```
#define RESTCONF_OPENSSL_NONBLOCKING 1
```

**Comment by olofhagsand** [comment]
Unfortunately we dropped regression for macos some years ago

**Comment by brunorijsman** [comment]
Closing as evidently macos is no longer supported.

**Comment by olofhagsand** [comment]
Well, we could add it again if there is interest and it is easy to maintain. 
The reason for dropping it was trivial if I remember correctly.
Are you using clixon on macos?

**Comment by brunorijsman** [comment]
My development environment is macOS but my target deployment environment is Ubuntu 22.04.   I already modified my development environment to start using Docker containers, so am able to proceed with development.  It would be nice if Clixon worked on macOS, but I am not blocked by it anymore.   Since my switch to Ubuntu, I did find another issue, namely https://github.com/clicon/clixon/issues/543

Just as background information: my website qukaydee.com offers a free service which is a cloud-hosted simulated quantum key distribution (QKD) network. It currently offers the ETSI QKD 014 API, which is a REST API.  I am in the process of implementing ETSI QKD 015 API, which is a YANG API.

**Comment by olofhagsand** [comment]
Cool, 
There are others using macos, we can always include your patches if you PR them, the problem for me is to run through regressions on a macos, so I need someone to help there.

**Comment by krihal** [comment]
Potential solution, does this look acceptable?

```
diff --git a/apps/restconf/restconf_main_native.c b/apps/restconf/restconf_main_native.c
index 385529cc6..a20c95b13 100644
--- a/apps/restconf/restconf_main_native.c
+++ b/apps/restconf/restconf_main_native.c
@@ -158,6 +158,11 @@
 #include "restconf_http1.h"
 #endif

+#ifndef SOCK_NONBLOCK
+#include <fcntl.h>
+#define SOCK_NONBLOCK O_NONBLOCK
+#endif
+
 /* Command line options to be passed to getopt(3) */
 #define RESTCONF_OPTS "hVD:f:E:l:C:p:y:a:u:rW:R:t:o:"
```

**Comment by brunorijsman** [comment]
Verified; fix works.
